packages/compile-solidity/src/run.ts

```diff
@@ -1,14 +1,81 @@
-const debug = require("debug")("compile:run");
-const OS = require("os");
-const semver = require("semver");
-const Common = require("@truffle/compile-common");
-const { CompilerSupplier } = require("./compilerSupplier");
-const { zeroLinkReferences, formatLinkReferences } = require("./shims");
+import { zeroLinkReferences, formatLinkReferences } from "./shims";
+import debugModule from "debug";
+const debug = debugModule("compile:run");
+import OS = require("os");
+import semver from "semver";
+import { CompilerSupplier } from "./compilerSupplier";
+import type {
+  Compilation,
+  Source,
+  CompiledContract
+} from "@truffle/compile-common";
+import type Config from "@truffle/config";
+import * as Common from "@truffle/compile-common";
+
+type Targets = string[];
+
+type CompilerOutput = {
+  contracts: {
+    [path: string]: object;
+  };
+  sources: {
+    [path: string]: {
+      ast?: object;
+      legacyAST?: object;
+      id: number;
+    };
+  };
+  errors?: any[];
+};
+
+type ProcessAllSourcesArgs = {
+  sources: Common.Sources.Sources;
+  compilerOutput: CompilerOutput;
+  originalSourcePaths: any;
+  language: string;
+};
+
+type PrepareCompilerInputArgs = {
+  sources: Common.Sources.Sources;
+  targets: Targets;
+  language: string;
+  settings: any;
+  modelCheckerSettings: any;
+};
+
+type InternalOptions = {
+  language?: string;
+  noTransform?: boolean;
+  solc?: any;
+};
+
+type PreparedSources = {
+  [path: string]: {
+    content: string;
+  };
+};
+
+type ProcessContractsArgs = {
+  compilerOutput: CompilerOutput;
+  sources: Common.Sources.Sources;
+  originalSourcePaths: Common.Sources.PathMapping;
+  solcVersion: string;
+};
+
+type Contracts = {
+  [path: string]: {
+    [name: string]: any;
+  };
+};
```

Comment:
Minor nit - you might consider moving these to a local `types.ts` file, which would be in line with other TS packages in the repo.